packages/carbon-graphs/src/js/controls/PairedResult/PairedResult.js

```diff
@@ -51,18 +51,36 @@ import { validatePairedResultData } from '../../helpers/constructUtils';
  *
  * @private
  * @param {Array} values - Datapoint values
- * @param {string} axis - y or y2
  * @returns {object} - Contains min and max values for the data points
  */
-const calculateValuesRange = (values, axis = constants.Y_AXIS) => ({
-  [axis]: {
-    min: Math.min(
-      ...values.map((i) => Math.min(...Object.keys(i).map((j) => i[j].y))),
-    ),
-    max: Math.max(
-      ...values.map((i) => Math.max(...Object.keys(i).map((j) => i[j].y))),
-    ),
-  },
+const calculateValuesRangeYAxis = (values) => ({
+  min: Math.min(
+    ...values.map((i) => Math.min(...Object.keys(i).map((j) => i[j].y))),
+  ),
+  max: Math.max(
+    ...values.map((i) => Math.max(...Object.keys(i).map((j) => i[j].y))),
+  ),
+});
+
+/**
+ * @typedef {object} PairedResult
+ * @typedef {object} GraphContent
+ * @typedef {object} PairedResultConfig
+ */
+/**
+ * Calculates the min and max values for X Axis
+ *
+ * @private
+ * @param {Array} values - Datapoint values
+ * @returns {object} - Contains min and max values for the data points
+ */
+const calculateValuesRangeXAxis = (values) => ({
+  min: Math.min(
+    ...values.map((i) => Math.min(...Object.keys(i).map((j) => i[j].x))),
+  ),
+  max: Math.max(
+    ...values.map((i) => Math.max(...Object.keys(i).map((j) => i[j].x))),
+  ),
```

Comment:
I would optimize this so that `...Object.keys(i).map((j) => i[j].x)` isn't executed twice by using a constant. Same for `calculateValuesRangeYAxis`.

Reply:
[323e8c5](https://github.com/cerner/terra-graphs/pull/289/commits/323e8c5351a94855f7ff69362b4374d1a5d19be8)